src/jobs/schemas/job.schema.ts

```diff
@@ -71,17 +82,17 @@ export class JobClass extends OwnableClass {
   })
   statusMessage: string;
 
-  // history of status codes
-  @ApiProperty({
-    type: [Object],
-    required: false,
-    description: "Array of status updates containing status code and message",
-  })
-  @Prop({
-    type: [Object],
-    required: false,
-  })
-  statusHistory: Record<string, string>[];
+  // // history of status codes
```

Comment:
Why is this commented out? Is it not needed?

Reply:
I would be prefer to keep this here, instead of a new collection. If we create a new collection then for each create and for each update we will have to modify two collections. Also imagine if we have a large amount of jobs that we will also have another collection of the same size just to store the status history.

Reply:
Sounds reasonable, although what if we have a job that has too many update and we reach the 16MB of the document in mongodb?

Reply:
Good point. Has it ever happened to reach the limit? Can we somehow calculate the average size of jobs document? I cannot really picture if for our use case 16MB is enough or not

Reply:
we reuse history class for a collection to store history for both jobs and datasets. we start with jobs first

Reply:
to merge this PR: we keep statusHistory in the jobs schema. Then create a separate PR to split it